src/crimson/os/seastore/transaction.h

```diff
@@ -471,7 +497,7 @@ class Transaction {
    * Contains a reference (without a refcount) to every extent mutated
    * as part of *this.  No contained extent may be referenced outside
    * of *this.  Every contained extent will be in one of inline_block_list,
-   * ool_block_list, mutated_block_list, or delayed_alloc_list.
+   * written_ool_block_list, mutated_block_list, or delayed_alloc_list.
```

Comment:
or pre_alloc_list, right?

Reply:
"or/and pre_alloc_list" seems to be more accurate according to https://github.com/ceph/ceph/pull/47969#discussion_r1023588427

Reply:
Does that behavior match the delayed_alloc_list?

Reply:
Please see the following explanation.
Segment ool: delayed_alloc_list ---after ool write---> written_ool_block_list
Random block ool: pre_alloc_list ---after ool write---> written_ool_block_list and pre_alloc_list

To simplify free logic when a conflict occurs---Note that we need to free allocated paddr in RBM case---, 
pre_alloc_list is not cleared at `get_valid_pre_alloc_list`.
Here is what we discussed (https://github.com/ceph/ceph/pull/47969#discussion_r1017482853); pre_alloc_list is formerly allocated_ool_block_list.
By doing so, we are able to just clear invalid extents in pre_alloc_list.

Reply:
> Does that behavior match the delayed_alloc_list?

Not exactly, there is a requirement to cancel allocations with transaction invalidation at any time. So, unlike `delayed_alloc_list`, the `pre_alloc_list` is preserved after adding to the `written_ool_block_list` in order to simplify the cancellations.